crates/oxc_linter/src/rules/no_compare_neg_zero.rs

```diff
@@ -0,0 +1,120 @@
+use oxc_ast::{
+    ast::{Expression, UnaryOperator},
+    AstKind, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-compare-neg-zero): Do not use the {0} operator to compare against -0.")]
+#[diagnostic(severity(warning))]
```

Comment:
I think we should have some kind of help here, why don't we want to compare against -0?

Reply:
Added help message. Does it look good to you?